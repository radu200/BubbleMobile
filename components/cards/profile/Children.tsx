import React from "react";
import { StyleSheet, Text, View, Image } from "react-native";
import StaticData from '../../../data/static'
import GirlIcon from "../../../assets/images/girl.png";
import BoyIcon from "../../../assets/images/boy.png";

type children = {
  age: number,
  gender: number
}
type props = {
  data: Array<children>
}


export default function Children({ data }: props) {
  const {
    children,
  } = StaticData.profile
  return (
    <>
      <Text style={styles.title}>{children.title}</Text>
      <View style={styles.childrenSection}>
        {data.map((c: children, i: number) => (
          <View key={i} style={styles.childrenContainer}>
            <Image style={styles.childrenIcon} source={c.gender ? BoyIcon : GirlIcon} />
            <Text>{c.age} {children.years} </Text>
          </View>
        ))}
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  title: {
    fontSize: 16,
    fontWeight: "bold",
    paddingBottom: 5,
  },
  childrenSection: {
    flexDirection: "row",
    flexWrap: 'wrap'
  },
  childrenContainer: {
    flexDirection: "column",
    alignItems: 'center',
    justifyContent: 'center',
    margin: 5
  },
  childrenIcon: {
    width: 25,
    height: 25,
    margin: 5,
  }
});
