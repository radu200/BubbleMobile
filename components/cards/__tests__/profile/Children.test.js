import React from 'react';
import renderer from 'react-test-renderer';
import Children from '../../profile/Children';

it('Children component renders correctly', () => {
    const data = [{
        age: 2,
        gender: 0
    },
    {
        age: 11,
        gender: 1
    }]
    const tree = renderer
        .create(<Children data={data} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})