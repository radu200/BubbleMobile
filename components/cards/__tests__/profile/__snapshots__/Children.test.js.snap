// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children component renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "paddingBottom": 5,
      }
    }
  >
    Children
  </Text>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 5,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "margin": 5,
            "width": 25,
          }
        }
      />
      <Text>
        2
         
        years
         
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 5,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "margin": 5,
            "width": 25,
          }
        }
      />
      <Text>
        11
         
        years
         
      </Text>
    </View>
  </View>,
]
`;
